### PR TITLE
Add 6 Dynamic Programming questions and their solutions.

### PR DESCRIPTION
**Pull request by sinhatushar**
As a part of KWOC, I am adding 6 dynamic programming questions in the folder of dynamic programming. I will keep on adding from now on.It took me a while to understand the way Github works and hence my first commit is coming a bit late.

### CODE REVIEW

File: Dynamic programming - algos/binomial_coeff.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
+
```
Comment (khanchi97) [inline Dynamic programming - algos/binomial_coeff.cpp:1]: @sinhatushar Please use `/*` for start comments in multiple lines and `*/` for ending.And please also make these code to prompt the user to ask what he has to enter:)

File: Dynamic programming - algos/path_in_matrix.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+//////
+
+
+////////////////////////////Solution/////////////////////////////
+
+#include <bits/stdc++.h>
+using namespace std;
+
+
+int max_sum(vector<vector<int>> matrix)
```
Comment (22shubh22) [inline Dynamic programming - algos/path_in_matrix.cpp]: This is not a good way to pass vector to a function (here actually vector is copied). have a look at http://www.geeksforgeeks.org/passing-vector-function-cpp/  . Also between two '>>' it should be space seperated (as confuse it with bit manupulation.)

File: Dynamic programming - algos/path_in_matrix.cpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+{
+	vector<vector<int>> dp(matrix.size(),vector<int>(matrix.size(),0));
```
Comment (22shubh22) [inline Dynamic programming - algos/path_in_matrix.cpp]: Space between "<<".

File: Dynamic programming - algos/path_in_matrix.cpp [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            for(int j=0;j<n;j++)
+	    	{
+	                  if(i==0)
+	                  {
+	                  	dp[i][j]=matrix[i][j] ;
+	                  }
+	                  else if(j==0)
+	                  {
+	                  	dp[i][j]=max(dp[i-1][j],dp[i-1][j+1]) + matrix[i][j] ;
+	                  }
```
Comment (22shubh22) [inline Dynamic programming - algos/path_in_matrix.cpp]: Include boundary condition when ( j==n-1 ) dp[i][j]= max(dp[i-1][j-1], dp[i-1][j]).

File: Dynamic programming - algos/max_sub_square.cpp [hunk 5]
```diff
@@ -0,0 +1,79 @@
+////////////////////////////Problem statement/////////////////////////////
+
+/////
+
+Given a binary matrix, find out the maximum size square sub-matrix with all 1s.
+
+Input:
+
+The first line of input contains an integer T denoting the number of test cases.
+The first line of each test case is n and m,n is the number of rows and m is the number of columns.
```
Comment (22shubh22) [inline Dynamic programming - algos/max_sub_square.cpp:10]: there are m rows and n columns. ( Reference line 65 - 75, study line 73 carefully). Also edit the information then in line no. 10,11,21.

File: Dynamic programming - algos/longest_common_subsequence.cpp [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+////////////////////////////Solution///////////////////////////////////
+
+
+#include <bits/stdc++.h>
+using namespace std;
+
+int LCS(string A,string B)
+{
+    vector< vector<int>> dp(A.length()+1,vector<int>(B.length()+1));
```
Comment (22shubh22) [inline Dynamic programming - algos/longest_common_subsequence.cpp]: Space should be there between '>>'.

File: Dynamic programming - algos/binomial_coeff.cpp [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+#include <bits/stdc++.h>
+using namespace std;
+
+
+int nCr(int n, int r)
+{
+    int M=1000000007;
+    vector< vector<int>> dp (n+1,vector<int>(r+1)) ;
```
Comment (22shubh22) [inline Dynamic programming - algos/binomial_coeff.cpp]: space should present between '>>'

**Review by 22shubh22 — CHANGES_REQUESTED** [review]
Make some small changes.

File: Dynamic programming - algos/binomial_coeff.cpp (hunk 1 above)
Comment (sinhatushar) [reply Dynamic programming - algos/binomial_coeff.cpp:1]: Changes have been made as per your and @22shubh22 's review.

**Review by 22shubh22 — APPROVED** [review]
Well done sintushar!! Looking forward for more contributions from you.

**Pull request merged**